nats-base-client/heartbeats.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { deferred, Deferred } from "./util.ts";
+import { DebugEvents, Status } from "./types.ts";
+
+export interface PH {
+  flush(p?: Deferred<void>): Promise<void>;
+  disconnect(): void;
+  dispatchStatus(status: Status): void;
+}
+
+export class Heartbeat {
+  ph: PH;
+  interval: number;
+  maxOut: number;
+  timer?: number;
+  pendings: Promise<void>[] = [];
+
+  constructor(ph: PH, interval: number, maxOut: number) {
+    this.ph = ph;
+    this.interval = interval;
+    this.maxOut = maxOut;
+  }
+
+  // api to start the heartbeats, since this can be
+  // spuriously called from dial, insure we don't
+  // leak timers
+  start() {
+    this.cancel();
+    this._schedule();
+  }
+
+  // api for canceling the heartbeats, if a string is
```

Comment:
I don't understand `if a string is provided..` where?

Reply:
stale comment.

Reply:
fixed